jena-core/src/main/java/com/hp/hpl/jena/datatypes/xsd/XSDDatatype.java

```diff
@@ -251,7 +269,7 @@ public XSDDatatype(String typeName, Class<?> javaClass) {
      * @param xstype the XSSimpleType definition to be wrapped
      * @param namespace the namespace for the type (used because the grammar loading doesn't seem to keep that)
      */
-    public XSDDatatype(XSSimpleType xstype, String namespace) {
+    XSDDatatype(XSSimpleType xstype, String namespace) {
```

Comment:
Jena provides all the meaningful XSD types (just some XML specific types are missing) so it is unlikely that external code is calling this public constructor.  Howver, it is public.  If the aim is to hide Xerces XSSimpleType, then hop 
about:

1. Make this protected.
2. Add `public XSDDatatype(Object xstype, String namespace)` and cast to `XSSimpleType`

I thhnk that will keep OSGi happy(ier).